src/_pytest/pythonpath.py

```diff
@@ -0,0 +1,22 @@
+import sys
+
+import pytest
+from pytest import Config
+from pytest import Parser
+
+
+def pytest_addoption(parser: Parser) -> None:
+    parser.addini("pythonpath", type="paths", help="Add paths to sys.path", default=[])
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_load_initial_conftests(early_config: Config) -> None:
+    # `pythonpath = a b` will set `sys.path` to `[a, b, x, y, z, ...]`
+    for path in reversed(early_config.getini("pythonpath")):
+        sys.path.insert(0, str(path))
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_unconfigure(config: Config) -> None:
+    for path in config.getini("pythonpath"):
+        sys.path.remove(path)
```

Comment:
This needs to be `sys.path.remove(str(path))`. I guess we do need to be able to test it somehow... I wonder how the existing tests didn't catch it. If you can't make it work, I can try and find a way.

Also, I figure if the test suite itself somehow mangles the sys.path and removes one of the paths we added, we shouldn't crash. So I think it would be a good idea to wrap the `sys.path.remove` with `try: ... except ValueError: pass`.

Reply:
fixed the str. Also not sure how the existing tests didn't catch it. I'll look at that more.
Regarding try/except, I'm not sure how to test the except case, so I changed code to `if p in sys.path: remove`
I'm not sure how to test that the end path is correct.